salalib/alllinemap.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include "salalib/axialmap.h"
+#include "salalib/axialpolygons.h"
+
+class AllLineMap: public ShapeGraph
+{
+public:
+    AllLineMap(Communicator *comm,
```

Comment:
I don't particularly like the structure of this all line map - it does three things in a weird mix:
- on construction, it actually calculates an All Line Map. 
- it has some extra data that is required for an all line map, but that is public and can be accessed for read and write from external. I am assuming that the constructor that just takes a name is used to create an empty all line map and fill it with data from externally when de-serialising?
- it provides methods to reduce an all line map to a fewest line map.

I am wondering whether we wouldn't be better off with a data structure that just holds data, and functions to do the calculations?